app/src/test/java/org/simple/clinic/enterotp/EnterOtpScreenControllerTest.kt

```diff
@@ -387,100 +390,166 @@ class EnterOtpScreenControllerTest {
-        SmsSendShowMessageTestParams(result = ActivateUser.Result.ServerError(400), shouldShowMessage = false)
-    )
+    verify(screen).hideProgress()
   }
 
   @Test
-  @Parameters(method = "params for show error on request otp")
-  fun `when the resend sms call completes, the error must be shown`(params: SmsSendShowErrorParams) {
+  fun `when the resend sms call succeeds, the sms sent message must be shown`() {
     whenever(activateUser.activate(loggedInUserUuid, pin))
-        .doReturn(params.result)
+        .doReturn(ActivateUser.Result.Success(userPayload))
     whenever(userSession.loggedInUserImmediate())
         .doReturn(TestData.loggedInUser(uuid = loggedInUserUuid))
     whenever(ongoingLoginEntryRepository.entryImmediate())
         .doReturn(ongoingLoginEntry)
 
     uiEvents.onNext(EnterOtpResendSmsClicked())
 
-    params.verifications.invoke(screen)
+    verify(screen).showSmsSentMessage()
   }
 
-  data class SmsSendShowErrorParams(
-      val result: ActivateUser.Result,
-      // FIXME 25-Jul-19: This should be broken into separate tests
-      val verifications: (Ui) -> Unit
-  )
+  @Test
+  fun `when the resend sms call fails with a network error, the sms sent message must not be shown`() {
+    whenever(activateUser.activate(loggedInUserUuid, pin))
+        .doReturn(ActivateUser.Result.NetworkError)
+    whenever(userSession.loggedInUserImmediate())
+        .doReturn(TestData.loggedInUser(uuid = loggedInUserUuid))
+    whenever(ongoingLoginEntryRepository.entryImmediate())
+        .doReturn(ongoingLoginEntry)
 
-  @Suppress("Unused")
-  private fun `params for show error on request otp`(): List<SmsSendShowErrorParams> {
-    return listOf(
-        SmsSendShowErrorParams(
-            result = ActivateUser.Result.Success(userPayload),
-            verifications = { screen: Ui ->
-              verify(screen, never()).showNetworkError()
-              verify(screen, never()).showServerError(any())
-              verify(screen, never()).showUnexpectedError()
-            }
-        ),
-        SmsSendShowErrorParams(
-            result = ActivateUser.Result.NetworkError,
-            verifications = { screen: Ui -> verify(screen).showNetworkError() }
-        ),
-        SmsSendShowErrorParams(
-            result = ActivateUser.Result.OtherError(RuntimeException()),
-            verifications = { screen: Ui -> verify(screen).showUnexpectedError() }
-        ),
-        SmsSendShowErrorParams(
-            result = ActivateUser.Result.ServerError(400),
-            verifications = { screen: Ui -> verify(screen).showUnexpectedError() }
-        )
-    )
+    uiEvents.onNext(EnterOtpResendSmsClicked())
+
+    verify(screen, never()).showSmsSentMessage()
+  }
+
+  @Test
+  fun `when the resend sms call fails with a server error, the sms sent message must not be shown`() {
+    whenever(activateUser.activate(loggedInUserUuid, pin))
+        .doReturn(ActivateUser.Result.ServerError(400))
```

Comment:
Should the `ServerError` code be 500 instead or 400 is fine?

Reply:
Good point; Although it makes no difference to the tests, we should strive to match real world scenarios. I'll make this change in a later PR.